CSS styling update and auto-saving settings

mdiller opened the pull request:
I was intending to just clean up some css styling stuff but then had some other ideas and got a bit carried away. Feel free to let me know if any of the things listed below should be changed/removed.

(Note: Pull request was re-created as **Madrang** let me know in discord that i should be based on and merging into the beta branch)

![image](https://user-images.githubusercontent.com/3231343/195958664-46604f5f-ce85-4a47-8f8c-0fec89dd9ed4.png)


# Changes
- Updated the "Image Settings" input form section to make it more readable
- Updated the style of all of the input controls to match
- Darkened the style of the input controls
  - Note that i left checkboxes, silders, and buttons alone for the most part. I could update the styling for these too, but it'd be more involved and im not sure what people would want.
  - Also updated the colors for checkboxes and sliders to use the purple we're using for the "Make Image" button
- Added/used the "Work Sans" font
- Added some additional styling to the image uploaded img2img
  - Added rounded corners 
  - It will now will show the aspect ratio correctly
  - Added a little note to the bottom left of the image that displays the "width x height"
  - If you hover on the image for more than 0.5 seconds, it will grow larger (for easier comparison to the generated results)
- Added support for auto-saving settings. (prompt + things in the "Image Settings" section)
  - Turning on/off this feature is doable from the system-settings section
    - A checkbox is there to disable/enable the feature
    - A button is there to reset settings to default
    - Another button is there to bring up the configuration
  - You can bring up a window to configure which settings are persisted
  - Currently the behavior is really that it always saves and just doesnt *load* if the user hasn't enabled the feature. I figured that'd result in less confusing behavior. Let me know if you want me to change that to not save anything unless the user checks the box.
  - The way its implemented, it should be really easy to add new settings to this if new ones get added in the future, just add it to the `SETTINGS_TO_SAVE` list and it should get added smoothly
  - It currently doesn't support auto-saving the image loaded for img2img or any of the "Image Modifiers", but it would be possible to add support for that if its wanted
  - I could also at some point add support for saving different presets for settings if that is wanted


I'm happy to revert/remove any of the changes in the above list, I figured some of them like the dark controls or the font might not be desired. Just let me know if we want to keep some of them but not all of them and I can update this PR. If we want dark/light mode but want the option to swap between them let me know and i can implement that too. I can also split this up into 2 PRs (one for the visual updates and one for the auto-saving settings) if you want that instead.

Let me know if you want anything changed!

## Images
Heres an image that shows the majority of the changes with a comparison to the old version

![image](https://user-images.githubusercontent.com/3231343/195958664-46604f5f-ce85-4a47-8f8c-0fec89dd9ed4.png)

Click on any of the below things to un-hide the images
<details>
  <summary>Without the dark coloring</summary>

![image](https://user-images.githubusercontent.com/3231343/195958687-91eadd9c-21b5-4385-aa37-cac58a1f3926.png)

</details>
<details>
  <summary>After hovering over the img2img image for 0.5s</summary>

![image](https://user-images.githubusercontent.com/3231343/195958750-0acd640a-0724-4126-bee5-360e085475b2.png)

</details>
<details>
  <summary>The new entry in the system settings</summary>
Unchecked:

![image](https://user-images.githubusercontent.com/3231343/195958833-eede0b80-e91a-4004-bf33-30afad09e665.png)

Checked:

![image](https://user-images.githubusercontent.com/3231343/195958842-4b927089-d0c4-42b2-aa47-ca223a663cfa.png)

</details>
<details>
  <summary>The configuration screen that comes up when you hit the "Configure" button</summary>

![image](https://user-images.githubusercontent.com/3231343/195958858-3f0988d6-0ff8-4937-b8d3-2488efc36a32.png)

</details>

cmdr2 commented:
Nice work! I'm quite okay with this. Just two comments:
1. Like you suggested, can we please split the auto-save into a separate PR? From the code changes, it looks fairly easy to separate the two.
2. I apologize for the slightly extra work, but can you please move the auto-save settings related JS changes to a separate file, like `media/auto-save.js`? And include that JS file just before `main.js` is included in index.html.

I'd suggest moving the `getLocalStorageBoolItem()` function, and probably even all the other settings saving code (the existing ones) into this `auto-save.js` file. That's why I apologized, because it's a bit of extra work for you :) But I'm really trying to refactor the codebase gradually into more manageable chunks, so your help here would be greatly appreciated!

Thanks again for this change, I like it.

mdiller commented:
sounds good to me! one question for when I work on this tomorrow: when I migrate those other settings persisting stuff over to this new file, do you think it'd be a good idea to just merge it with the new system i made for saving settings or keep it separate?

mdiller commented:
Ok I've removed all the styling stuff from this branch and moved the auto-save logic into a separate file. I haven't touched the `getLocalStorageBoolItem()` stuff *yet*, but I like the idea of moving it to the new file to have it live with the other settings stuff too. I'll wait on moving it till I hear back from you on if its ok to rework those settings into the new system. Gonna go work on the styling part now to fix it up a bit and put it in its own branch.

cmdr2 commented:
I messed up the merge, and have fixed it by taking the changes and committing it separately (credited you in the commit msg) - https://github.com/cmdr2/stable-diffusion-ui/commit/188894c837d5d48a036249dcd32e0afe94b00b1e

Basically I didn't see that the source branch didn't contain the styling changes. So when I merged this PR after I had merged the styling changes, this merge reverted all the styling changes. Things got a bit messy while I tried to fix this, and finally it was just simpler to recreate your PR and commit it separately.

Thanks for the PR! Looks good
